src/array.jl

```diff
@@ -972,3 +972,31 @@ Base.repeat(a::CatArrOrSub{T, N},
 Base.repeat(a::CatArrOrSub{T, N};
             inner = nothing, outer = nothing) where {T, N} =
     CategoricalArray{T, N}(repeat(refs(a), inner=inner, outer=outer), copy(pool(a)))
+
+# JSON3 writing/reading
+StructTypes.StructType(::Type{<:CategoricalVector}) = StructTypes.ArrayType()
+
+StructTypes.construct(::Type{<:CategoricalVector}, array::Vector) =
+    categoricalgeneral(array)
+StructTypes.construct(::Type{<:CategoricalArray}, array::Vector) =
+    categoricalgeneral(array)
+function categoricalgeneral(array)
+    if eltype(array) >: Nothing
+        categorical(replace(array, nothing=>missing))
+    else
+        categorical(array)
+    end
+end
+
+StructTypes.construct(::Type{<:CategoricalVector{Union{Missing, T}}},
```

Comment:
@nalimilan does the catch-all subtyping operator do anything here? or is `CategoricalVector{Union{Missing, T}}` fully specified?

Reply:
There are more parameters that aren't specified here.